### DISCUSSION TITLE
Spring AMQP tracking feature ?

### DISCUSSION THREAD

**Discussion by dimitri995** [open-telemetry/opentelemetry-java-instrumentation#12355] [Q&A] [answered]
Hi all,

I have a spring application, and I am using spring amqp, is it possible to instrument it without java agent ? There is a library ?

Thanks

Dimitri

**Comment by laurit** [comment, marked as answer]
I don't think we currently have manual instrumentation for this. We have agent instrumentation for `spring-rabbit` https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/spring/spring-rabbit-1.0/javaagent Perhaps a manual instrumentation library could be extracted from it. There is also https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/10091 that attempts to build a manual instrumentation library for rabbitmq.
